Update: Make radix accept a "as-needed" option (fixes #4048)

mzgol opened the pull request:
This makes the radix rule disallow using the second parameter to parseInt
when it's 10. Other values are allowed.

BTW, I'm not sure if `"never"` is a good name as it suggests radix is disallowed in all cases but here we only disallow the exact `10` value (in the future we might want to also include variables statically-determinably equal to 10 but that's another story). Perhaps `"no10"` would be a better name? Or `"disallow10"`, `"forbid10"`?

silverwind commented:
Maybe `no-decimal`?

mzgol commented:
@silverwind I'm *really* bad with names, your suggestion sounds nice!

nzakas commented:
You're right, never isn't the best word. We typically use "as-needed" in cases like this, which is to say that there's a default and you only need the extra param if you don't want the default.

silverwind commented:
+1 to `as-needed`

mzgol commented:
PR updated & rebased. I also changed the message for parameter-less `parseInt()` - it used to say "Missing radix parameter" but I realised this one should be rejected even with the `as-needed` option where this message would be incorrect so a separate one was needed anyway. For consistency I changed it in both cases, it now says `"No parameters passed."`.

gyandeeps commented:
I think we are missing documentation update for this new option.

mzgol commented:
You're right, I forgot about that. I can add it early next week.



-- 
Michał Gołębiowski

nzakas commented on lib/rules/radix.js:
```diff
@@ -25,17 +30,37 @@ module.exports = function(context) {
+            if (node.arguments.length === 0) {
+                context.report({
+                    node: node,
+                    message: "No parameters passed."
```
I'd say "Missing parameters" to better align with the other messages.

nzakas commented:
@mzgol can you address that last comment?

mzgol commented:
I'm sorry, I got busy back then and I couldn't have finished it. I'll try
to do it this week.

nzakas commented:
I'll make the last change.

nzakas commented:
Closing in favor of #4351
